src/modules/controllers/user.cotroller.js

```diff
@@ -1 +1,29 @@
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
 const Users = require('../../db/models/users/index');
+
+const jwtSecret = 'xkslfm29irj3rtf2m3fdio';
+
+module.exports.newUser = async (req, res) => {
+  const { login, password } = req.body;
+  if (login && password) {
+    const isUsed = await Users.findOne({ login });
+    if (isUsed) {
+      return res.status(300).json({ message: 'данный эмайл уже занят' });
+    }
+    const hashedPassword = await bcrypt.hash(password, 10);
+    const user = new Users({
+      login, password: hashedPassword,
+    });
+    const token = jwt.sign(
+      { userId: user.id },
```

Comment:
where did you get the id of a user that has not yet been saved?

Reply:
```
const user = new Users({
      login, password: hashedPassword,
    });
```
this variable create new user and get user id. and i use it and send `token` on client. if client response token, he writes token in local storage and redirects on main page. user id exist when i use it.